File: Python/Communication entre processus/4.1.py
import os

# Création des deux pipes
p1_r, p1_w = os.pipe()   # Fils -> Père
p2_r, p2_w = os.pipe()   # Père -> Fils

pid = os.fork()

if pid == 0:
    word = input("Entrez un mot : ")

    # Envoi du mot au père
    os.write(p1_w, word.encode())

    # Lecture du résultat du père
    result = os.read(p2_r, 10).decode()

    print("Résultat :", result)

    os._exit(0)

else:

    # Lecture du mot envoyé par le fils
    word = os.read(p1_r, 100).decode()

    found = 0

    try:
        with open("file.txt", "r") as f:
            for line in f.read().split():
                if line == word:
                    found = 1
                    break
    except:
        print("Erreur ouverture fichier")
        os._exit(1)

    # Envoi du résultat au fils
    os.write(p2_w, str(found).encode())

    os.wait()

File: C/Communication entre processus/4.1.c
#include<stdio.h>
#include<unistd.h>
#include<string.h>
#include<stdlib.h>
#include <sys/wait.h>

int main() 
{
    int p1[2] , p2[2];
    pipe(p1);
    pipe(p2);

    pid_t pid = fork();

    if(pid == 0)
    {
        char word[100]; 
        printf("Entrez un mot : ");
        scanf("%s", word);

        write(p1[1], word, strlen(word)+1);

        int results;
        read(p2[0], &results, sizeof(int));

        printf("Résultat : %d\n", results);

        exit(0);
    }
    else
    {   

        char word[100];
        read(p1[0], word, sizeof(word));

        FILE *f = fopen("file.txt", "r");

        if(f == NULL)
        {
            perror("Erreur ouverture fichier");
            exit(1);
        }

        char line[100];
        int found = 0;

        while(fscanf(f, "%s", line) != EOF)
        {
            if(strcmp(word, line) == 0)
            {
                found = 1;
                break;
            }
        }

        fclose(f);

        write(p2[1], &found, sizeof(int));

        wait(NULL);
    }

    return (0);
}